### ISSUE TITLE
Catastrophic Regex backtracking with default config for "completed-docs"-rule

### ISSUE THREAD

**Issue by zetamorph** [palantir/tslint#3880]
### Bug Report

- __TSLint version__: 5.10.0
- __TypeScript version__: 2.8.3
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
class Test {
    /**
    **/
    private async test(): Promise<void> { }
}
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "completed-docs": [ ]
    }
}
```

or:

```json
{
    "rules": {
        "completed-docs": [true]
    }
}
```

#### Actual behavior

Nothing happens for a long time(it would probably finish after a few hours, but I didn´t try), CPU load at 100%.

#### Expected behavior

TSLint completes linting and outputs an error because of missing documentation.

### Additional Info

I traced the error back to this line: [tagExclusion.ts:78](https://github.com/palantir/tslint/blob/ecab88c8d60a2d5a7ee4f2d879e66b950971760e/src/rules/completed-docs/tagExclusion.ts#L78)

If an empty array or just `[true]` is given to the "completed-docs"-rule, the default args are loaded.
The regex above fails to match `/**\n**/` easily due to backtracking, but matches `/**\n*/` or `/****/` just fine.

This only happens when this [default setting](https://github.com/palantir/tslint/blob/ecab88c8d60a2d5a7ee4f2d879e66b950971760e/src/rules/completedDocsRule.ts#L85) is loaded.

This bug shouldn´t be very common and we just stumbled on it by accident(and having an invalid tslint.json), but it was very time-intensive to figure out what´s even happening so it would be nice to have this resolved in case other people happen upon the same issue.

### More Code that crashes TSLint

```
class Test {
    /**
     * foo
    **/
    private async test(): Promise<void> {
    }
}
```

**Comment by giladgray** [comment]
to be fair, this closing `**/` is non-standard syntax: JSDoc comments explicitly begin with `/**` and close with `*/`. that said, we should still fix the regex. shouldn't be _too_ hard (he said carefully, knowing full well just how hard regexes can be 👣 🔫 ).

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is deprecated** and no longer accepting pull requests other than security fixes. See #4534. ☠️ 
We recommend you instead use **[typescript-eslint](https://typescript-eslint.io)** to lint your TypeScript code with **ESLint**. ✅

👋 It was a pleasure open sourcing with you!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
